File: src/Core/Logistics.Application.Tenant/Commands/Payment/ProcessPayment/ProcessPaymentCommand.cs
using Logistics.Shared.Consts;
using MediatR;

namespace Logistics.Application.Tenant.Commands;

public class ProcessPaymentCommand : IRequest<Result>
{
    public string PaymentId { get; set; } = default!;
    public PaymentMethod PaymentMethod { get; set; }
    public string? CardholderName { get; set; }
    public string? CardNumber { get; set; }
    public string? CardExpirationDate { get; set; }
    public string? CardCvv { get; set; }
    public string? BankName { get; set; }
    public string? BankAccountNumber { get; set; }
    public string? BankRoutingNumber { get; set; }
    public string BillingAddress { get; set; } = default!;
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/payment/ProcessPaymentCommand.ts
import {PaymentMethod} from '@/core/enums';
import {AddressDto} from '../AddressDto';


export interface ProcessPaymentCommand {
  paymentId: string;
  paymentMethod: PaymentMethod;
  cardholderName?: string;
  cardNumber?: string;
  cardExpirationDate?: string;
  cardCvv?: string;
  bankName?: string;
  bankAccountNumber?: string;
  bankRoutingNumber?: string;
  billingAddress: AddressDto;
}
